spec/app/github_repo_fetcher_spec.rb

```diff
@@ -82,39 +82,42 @@ def readme_url
   end
 
   describe "#docs" do
+    let(:repo_name) { SecureRandom.uuid }
+
     def docs_url(repo_name)
       "https://api.github.com/repos/alphagov/#{repo_name}/contents/docs"
     end
 
     it "returns an array of hashes including title derived from markdown contents" do
-      repo_name = SecureRandom.uuid
       markdown_url = "https://raw.githubusercontent.com/alphagov/#{repo_name}/master/docs/analytics.md"
       source_url = "https://github.com/alphagov/#{repo_name}/blob/master/docs/analytics.md"
       markdown_fixture = "# Analytics \n Foo"
-      api_response = [
-        {
-          name: "analytics.md",
-          download_url: markdown_url,
-          html_url: source_url,
-        },
-      ]
-      expected_output = [
+      path = "docs/analytics.md"
+      default_branch = "main"
+      latest_commit = { sha: SecureRandom.hex(40), timestamp: Time.now.utc.to_s }
+      doc_response = [{ name: "analytics.md", path: path, download_url: markdown_url, html_url: source_url }]
+      commit_response = [{ sha: latest_commit[:sha], commit: { author: { date: latest_commit[:timestamp] } } }]
+
+      allow(GitHubRepoFetcher.instance).to receive(:repo)
+        .with(repo_name) { OpenStruct.new(default_branch: default_branch) }
+      stub_request(:get, docs_url(repo_name))
+        .to_return(body: doc_response.to_json, headers: { content_type: "application/json" })
+      stub_request(:get, "https://api.github.com/repos/alphagov/#{repo_name}/commits?path=#{path}&per_page=100&sha=#{default_branch}")
```

Comment:
Minor: these stubs are getting quite hairy and numerous, so they make it harder to read through the test. Could we extract them to helper methods for the tests?